### PR TITLE
refactor(core): side bar resizing

### LINKED ISSUE #5271: TOC button sometimes appears and sometimes disappears
version: 0.10.4 beta

https://github.com/toeverything/blocksuite/assets/142964904/95b1a888-f1a0-4675-b85d-38ea6011a45d

### PR DESCRIPTION
**Pull request by pengx17**
Rewrite sidebar panel using a customized react-resizable-panels version that supports sidebar pixel sizing (not using flex percentages).

Now the left & right sidebar using the same `ResizePanel` impl.

fix https://github.com/toeverything/AFFiNE/issues/5271
fix TOV-163
fix TOV-146
fix TOV-168
fix TOV-109
fix TOV-165

### CODE REVIEW

**Comment by pengx17** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5287** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5287?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5280** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5280?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5284** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5284?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5280?utm_source=stack-comment).

**Review by Brooooooklyn — APPROVED** [review]

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 1]
```diff
@@ -67,16 +61,8 @@ interface DetailPageLayoutProps {
...
-function useEnableAnimation() {
-  const [enable, setEnable] = useState(false);
-  useEffect(() => {
-    window.setTimeout(() => {
-      setEnable(true);
-    }, 500);
-  }, []);
-  return enable;
-}
+const MIN_SIDEBAR_WIDTH = 320;
```
Comment (EYHN) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx:64]: https://github.com/toeverything/AFFiNE/blob/87cb68f7fb456644ffd949a7dc62693597969dbb/packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/editor-sidebar.css.ts#L9

need change default width to 320

**Review by EYHN — COMMENTED** [review]
I tried this pr locally. When I opened the sidebar and resized it, the width jumped from 300px to 320px.

**Comment by pengx17** [comment]
> I tried this pr locally. When I opened the sidebar and resized it, the width jumped from 300px to 320px.

will fix this in another pr

**Pull request merged**